src/SFML/Window/OSX/SFContext.mm

```diff
@@ -183,32 +199,48 @@
 
         // Antialiasing level
         attrs.push_back(NSOpenGLPFASamples);
-        attrs.push_back((NSOpenGLPixelFormatAttribute)settings.antialiasingLevel);
+        attrs.push_back((NSOpenGLPixelFormatAttribute)m_settings.antialiasingLevel);
 
         // No software renderer - only hardware renderer
         attrs.push_back(NSOpenGLPFAAccelerated);
     }
 
     // Support for OpenGL 3.2 on Mac OS X Lion and later:
-    // SFML 2 Graphics module uses some OpenGL features that are deprecated
-    // in OpenGL 3.2 and that are no more available with core context.
+    // SFML 2 Graphics module uses some OpenGL features that are deprecated in
+    // OpenGL 3.0 and that are no longer available in 3.1 and 3.2+ with a core context.
     // Therefore the Graphics module won't work as expected.
 
-    // 2.x are mapped to 2.1 since Apple only support that legacy version.
+    // 1.x/2.x are mapped to 2.1 since Apple only support that legacy version.
     // >=3.0 are mapped to a 3.2 core profile.
-    bool legacy = settings.majorVersion < 3;
+    bool legacy = m_settings.majorVersion < 3;
 
     if (legacy)
     {
+        m_settings.attributeFlags &= ~ContextSettings::Core;
+        m_settings.majorVersion = 2;
+        m_settings.minorVersion = 1;
         attrs.push_back(NSOpenGLPFAOpenGLProfile);
         attrs.push_back(NSOpenGLProfileVersionLegacy);
     }
     else
     {
+        if (!(m_settings.attributeFlags & ContextSettings::Core))
+        {
+            sf::err() << "Warning. Compatibility profile not supported on this platform." << std::endl;
+            m_settings.attributeFlags |= ContextSettings::Core;
+        }
+        m_settings.majorVersion = 3;
+        m_settings.minorVersion = 2;
         attrs.push_back(NSOpenGLPFAOpenGLProfile);
         attrs.push_back(NSOpenGLProfileVersion3_2Core);
```

Comment:
I've just read something saying that this flag might create a 4.x context on 10.9+ on some hardware. This implies that L291 might report slightly wrong information. But we could just close our eyes and pretend Apple never used a ***3_2*** flag for 4.x context...